### PR TITLE
Templatize KubeVirt namespace

### PR DESCRIPTION
**Pull request by atanasdinov**
- Extracted the namespace resource out of the operator manifest
- Chart is now installable via `helm install kubevirt <chart-repo> -n <namespace> --create-namespace`
- This should prevent errors in Rancher where a `kubevirt` namespace was specified but it was already part of the chart itself
- Upgrade and deletion flows are confirmed to be working properly

### CODE REVIEW

File: packages/kubevirt/charts/templates/namespace-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        - name: kubevirt-namespace-modifier
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: {{ .Values.kubecli.image }}
+          command:
+            - /bin/sh
+            - -c
+            - >
+              kubectl label namespace {{ .Release.Namespace }} kubevirt.io="";
+              kubectl label namespace {{ .Release.Namespace }} pod-security.kubernetes.io/enforce="privileged";
```
Comment (atanasdinov) [inline packages/kubevirt/charts/templates/namespace-hooks.yaml:71]: These are labels which the upstream manifests are shipped with. Unfortunately, there wasn't an easier way to do this than a pre-install hook... https://github.com/helm/helm/issues/3503#issuecomment-696712443

**Review by ipetrov117 — APPROVED** [review]
Changes look solid, I have concerns about the security implications that come with labelling a full namespace as `pod-security.kubernetes.io/enforce="privileged"` though..
Taking into account that this is something that comes from the upstream manifests, I'll approve this.

File: packages/kubevirt/charts/templates/namespace-hooks.yaml (hunk 1 above)
Comment (e-minguez) [reply packages/kubevirt/charts/templates/namespace-hooks.yaml:71]: My only concern is that everything related to this is a leftover that will live in the cluster for no reason. Honestly I don't know how to _fix_ it but maybe the container can remove "everything" after the namespace is labeled? I mean, running kubectl delete role kubevirt-namespace-modifier, etc.

Comment (atanasdinov) [reply packages/kubevirt/charts/templates/namespace-hooks.yaml:71]: They won't live forever, the delete policy drops them all immediately.

```
root@atanas:~/charts/charts# helm install kubevirt kubevirt/0.2.0 -n kubevirt --create-namespace --debug
install.go:192: [debug] Original chart version: ""
install.go:209: [debug] CHART PATH: /root/charts/charts/kubevirt/0.2.0

client.go:128: [debug] creating 1 resource(s)
install.go:151: [debug] CRD kubevirts.kubevirt.io is already present. Skipping.
client.go:128: [debug] creating 1 resource(s)
client.go:128: [debug] creating 1 resource(s)
client.go:128: [debug] creating 1 resource(s)
client.go:128: [debug] creating 1 resource(s)
client.go:128: [debug] creating 1 resource(s)
client.go:540: [debug] Watching for changes to Job kubevirt-namespace-modifier with timeout of 5m0s
client.go:568: [debug] Add/Modify event for kubevirt-namespace-modifier: ADDED
client.go:607: [debug] kubevirt-namespace-modifier: Jobs active: 0, jobs failed: 0, jobs succeeded: 0
client.go:568: [debug] Add/Modify event for kubevirt-namespace-modifier: MODIFIED
client.go:607: [debug] kubevirt-namespace-modifier: Jobs active: 1, jobs failed: 0, jobs succeeded: 0
client.go:568: [debug] Add/Modify event for kubevirt-namespace-modifier: MODIFIED
client.go:607: [debug] kubevirt-namespace-modifier: Jobs active: 0, jobs failed: 0, jobs succeeded: 0
client.go:568: [debug] Add/Modify event for kubevirt-namespace-modifier: MODIFIED
client.go:310: [debug] Starting delete for "kubevirt-namespace-modifier" ServiceAccount
client.go:310: [debug] Starting delete for "kubevirt-namespace-modifier" Role
client.go:310: [debug] Starting delete for "kubevirt-namespace-modifier" RoleBinding
client.go:310: [debug] Starting delete for "kubevirt-namespace-modifier" Job
```

Comment (e-minguez) [reply packages/kubevirt/charts/templates/namespace-hooks.yaml:71]: You can ignore my comment then :D Sorry for the noise.

**Pull request merged**